src/nvim/spell.c

```diff
@@ -1896,11 +1892,12 @@ void count_common_word(slang_T *lp, char_u *word, int len, int count)
   if (HASHITEM_EMPTY(hi)) {
     wc = xmalloc(sizeof(wordcount_T) + p_len);
     memcpy(wc->wc_word, p, p_len + 1);
-    wc->wc_count = count;
+    wc->wc_count = (uint16_t)count;
     hash_add_item(&lp->sl_wordcount, hi, wc->wc_word, hash);
   } else {
     wc = HI2WC(hi);
-    if ((wc->wc_count += count) < (unsigned)count) {    // check for overflow
+    wc->wc_count = (uint16_t)(wc->wc_count + count);
+    if (wc->wc_count < count) {    // check for overflow
```

Comment:
It may be better to change the type of `count` to `uint16_t`, and keep this `unsigned` cast.

Reply:
Wait what?

So the line should be

```c
    wc->wc_count = (uint16_t)(wc->wc_count + (unsigned)count);
```
?

Reply:
I mean the argument `count`